client/src/models/Test/Test.ts

```diff
@@ -54,9 +54,22 @@ class Test {
     this.store.set(this.getComponentFormat());
   }
 
+  calculateScore(): number {
+    let score = 0;
+    this.questions.forEach(question => {
+      let MCquestion = question as MultipleChoiceQuestion;
```

Comment:
Some questions will not be MultipleChoiceQuestions and therefore instead of casting every question to the MultipleChoiceQuestion type, we should use a similar approach to line 34 of this file. 

Will be adding TrueAndFalse and "LinkedQuestions" in the near future.